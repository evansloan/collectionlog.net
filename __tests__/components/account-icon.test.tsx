import renderer from 'react-test-renderer';

import AccountIcon from '@/components/account-icon';

describe(`${AccountIcon.displayName} component`, () => {
  const accountTypes: AccountType[] = [
    'NORMAL',
    'IRONMAN',
    'HARDCORE_IRONMAN',
    'ULTIMATE_IRONMAN',
    'GROUP_IRONMAN',
    'HARDCORE_GROUP_IRONMAN',
    'UNRANKED_GROUP_IRONMAN',
  ];

  test.each(accountTypes)('should render with "$accountType" icon', (accountType) => {
    const render = renderer
      .create(<AccountIcon accountType={accountType} />)
      .toJSON();
    expect(render).toMatchSnapshot();
  });
});
