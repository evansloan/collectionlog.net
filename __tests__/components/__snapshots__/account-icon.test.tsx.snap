// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountIcon component should render with "$accountType" icon 1`] = `
<img
  alt="NORMAL icon"
  className="icon-shadow"
  data-nimg="1"
  decoding="async"
  height={15}
  loading="lazy"
  onError={[Function]}
  onLoad={[Function]}
  src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
  srcSet="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
  style={
    {
      "color": "transparent",
    }
  }
  width={15}
/>
`;

exports[`AccountIcon component should render with "$accountType" icon 2`] = `
<img
  alt="IRONMAN icon"
  className="icon-shadow"
  data-nimg="1"
  decoding="async"
  height={15}
  loading="lazy"
  onError={[Function]}
  onLoad={[Function]}
  src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
  srcSet="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
  style={
    {
      "color": "transparent",
    }
  }
  width={15}
/>
`;

exports[`AccountIcon component should render with "$accountType" icon 3`] = `
<img
  alt="HARDCORE_IRONMAN icon"
  className="icon-shadow"
  data-nimg="1"
  decoding="async"
  height={15}
  loading="lazy"
  onError={[Function]}
  onLoad={[Function]}
  src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
  srcSet="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
  style={
    {
      "color": "transparent",
    }
  }
  width={15}
/>
`;

exports[`AccountIcon component should render with "$accountType" icon 4`] = `
<img
  alt="ULTIMATE_IRONMAN icon"
  className="icon-shadow"
  data-nimg="1"
  decoding="async"
  height={15}
  loading="lazy"
  onError={[Function]}
  onLoad={[Function]}
  src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
  srcSet="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
  style={
    {
      "color": "transparent",
    }
  }
  width={15}
/>
`;

exports[`AccountIcon component should render with "$accountType" icon 5`] = `
<img
  alt="GROUP_IRONMAN icon"
  className="icon-shadow"
  data-nimg="1"
  decoding="async"
  height={15}
  loading="lazy"
  onError={[Function]}
  onLoad={[Function]}
  src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
  srcSet="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
  style={
    {
      "color": "transparent",
    }
  }
  width={15}
/>
`;

exports[`AccountIcon component should render with "$accountType" icon 6`] = `
<img
  alt="HARDCORE_GROUP_IRONMAN icon"
  className="icon-shadow"
  data-nimg="1"
  decoding="async"
  height={15}
  loading="lazy"
  onError={[Function]}
  onLoad={[Function]}
  src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
  srcSet="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
  style={
    {
      "color": "transparent",
    }
  }
  width={15}
/>
`;

exports[`AccountIcon component should render with "$accountType" icon 7`] = `
<img
  alt="UNRANKED_GROUP_IRONMAN icon"
  className="icon-shadow"
  data-nimg="1"
  decoding="async"
  height={15}
  loading="lazy"
  onError={[Function]}
  onLoad={[Function]}
  src="/_next/image?url=%2Fimg.jpg&w=32&q=75"
  srcSet="/_next/image?url=%2Fimg.jpg&w=16&q=75 1x, /_next/image?url=%2Fimg.jpg&w=32&q=75 2x"
  style={
    {
      "color": "transparent",
    }
  }
  width={15}
/>
`;
